### PR TITLE
Fixes #26228 - Fail ping when Candlepin is suspended

### PR DESCRIPTION
**Pull request by jturel**
Current behavior:

- put candlepin in suspend mode with `systemctl stop qpidd` on your server.
- verify suspend mode: `curl -k https://$HOSTNAME:8443/candlepin/status`
- check `hammer ping` and see that the candlepin status is OK (it's actually not OK)

Apply the path for new behavior:

- repeat above steps
- observe `hammer ping` now reports the failure correctly
- start qpidd and make sure candlepin is OK via curl
- confirm the same with `hammer ping`

Feel free to advise on the messaging - I made it up.

### CODE REVIEW

**Comment by omkarkhatavkar** [comment]
@jturel : Look good. 

**Hammer Ping: ** 

```
[vagrant@centos7-hammer-devel ~]$ hammer ping
candlepin:      
    Status:          FAIL
    Server Response:
candlepin_auth: 
    Status:          FAIL
    Server Response:
```
**UI Error Message :** 

candlepin_ping_error_message

File: app/models/katello/ping.rb [hunk 1]
```diff
@@ -40,7 +38,10 @@ def ping_pulp_without_auth(service_result, capsule_id)
       def ping_candlepin_without_auth(service_result)
         url = SETTINGS[:katello][:candlepin][:url]
         exception_watch(service_result) do
-          backend_status(url, :candlepin)
+          result = backend_status(url, :candlepin)
+          if result[:mode] == 'SUSPEND'
+            fail _("Candlepin was in suspend mode")
+          end
```
Comment (jlsherrill) [inline app/models/katello/ping.rb]: Are there any other 'modes' we should worry about?

**Comment by jlsherrill** [comment]
Overall looks good, but has rubocop failures, and one question

File: app/models/katello/ping.rb (hunk 1 above)
Comment (jturel) [reply app/models/katello/ping.rb]: Looks like NORMAL, SUSPEND are all we have

Comment (jturel) [reply app/models/katello/ping.rb]: I guess we could check mode != NORMAL?

Comment (jlsherrill) [reply app/models/katello/ping.rb]: I think i might prefer that, although you might ask the candlepin team which they see as more future proof

Comment (jturel) [reply app/models/katello/ping.rb]: Sounds like that is the better way to go - updated

**Pull request merged**